3rdparty/optee/libutee/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+set(LIBUTEE_S_SRC
+  ${OE_TZ_LIBUTEE_SRC}/arch/arm/utee_syscalls_a64.S
+  ${OE_TZ_LIBUTEE_SRC}/arch/arm/gprof/gprof_a64.S)
+
+set(LIBUTEE_C_SRC
+  ${OE_TZ_LIBUTEE_SRC}/tee_api_property.c
+  ${OE_TZ_LIBUTEE_SRC}/abort.c
+ #${OE_TZ_LIBUTEE_SRC}/trace_ext.c
+  ${OE_TZ_LIBUTEE_SRC}/assert.c
+  ${OE_TZ_LIBUTEE_SRC}/base64.c
+ #${OE_TZ_LIBUTEE_SRC}/tee_api_arith.c
+  ${OE_TZ_LIBUTEE_SRC}/tee_api.c
+  ${OE_TZ_LIBUTEE_SRC}/tee_api_objects.c
+  ${OE_TZ_LIBUTEE_SRC}/tee_api_operations.c
+  ${OE_TZ_LIBUTEE_SRC}/tee_api_se.c
+  ${OE_TZ_LIBUTEE_SRC}/tee_api_panic.c
+  ${OE_TZ_LIBUTEE_SRC}/tee_tcpudp_socket.c
+  ${OE_TZ_LIBUTEE_SRC}/tee_socket_pta.c
+  ${OE_TZ_LIBUTEE_SRC}/arch/arm/user_ta_entry.c
+  ${OE_TZ_LIBUTEE_SRC}/arch/arm/utee_misc.c
+  ${OE_TZ_LIBUTILS_EXT_SRC}/snprintk.c
+  ${OE_TZ_LIBUTILS_EXT_SRC}/buf_compare_ct.c)
+
+add_library(oeuteeasm OBJECT ${LIBUTEE_S_SRC})
```

Comment:
I would encourage just defining the sources directly in the two `add_library` calls, otherwise a reader has to pick up on the two different variables and libraries where the difference is only `S` vs `C` (and that's easy to miss at a glance 😆).